### ISSUE TITLE
Enhancement: menu entry to swap HDF file as you do with ADF files

### ISSUE THREAD

**Issue by theshinyknight** [Mazamars312/Analogue-Amiga#42]
I think this is self-explanatory: it works great to have a single HDF as auto boot, but would be great to have an option to change that without the need to modify a JSON file, since the pocket has no onboard editor nor a console you can use for things like this.

Adding the capability to change the attached HDF file and rebooting, would allow us to swap images for the hard drive and make different versions of the drive. Thanks

**Comment by rsn8887** [comment]
I found a silly workaround, but it is tedious:
- I made multiple copies of the core, by duplicating the cores/mazamars312.Amiga folder to cores/mazamars312.AmigaWB for example. 
- I edited the core name in the core.json file cores/mazamars312.AmigaWB/core.json to be AmigaWB instead of Amiga
- I edited the data.json file cores/mazamars312.AmigaWB/data.json to mount my own Workbench .hdf.
- I duplicated the assets/amiga/mazamars312.Amiga folder to assets/amiga/mazamars312.AmigaWB so that my new AmigaWB core would find its mpu.bin file.

now, when I launch the Amiga core I can select change core and choose between Amiga and AmigaWB. If I choose the AmigaWB core, it boots my WB hdf, and if I choose the Amiga core it boots the standard AmigaVision.

I then repeated the process a few time to create core versions for my favorite hdfs.

**Comment by theshinyknight** [comment]
Very nice solution; for the time being it could work (although it has the downside of creating a lot of single entries if you have a lot of variants). Thanks!

**Comment by Mazamars312** [comment]
The sad thing is that the MPU does not handle changing of the HDD on the fly very well. Even after rebooting.

So, at this moment, I am leaving this "hardwired" to a JSON-only change for other images.

**Comment by lunchpacked130** [comment]
I was also hoping to change HDF files hrough the menu, especially since i needed more space for copious amounts of tracker music and some other games and apps etc.. but i realized it most likely would never happen, and i didn't want to bother with getting a third HDF file so i found another way..

### Jump to guide

### Tl;dr

Note! This can be a bit technical and/or frustrating, especially if you've never worked with workbench before.
I won't be covering things like how to setup WinUAE or improve transfer speeds or how functions of workbench work or what programs to use anything like that, I'm just covering the basics of how i got more than enough space for my needs.
But as long as you have backups, it doesn't hurt to play around and try it yourself, but this fix might not be for everyone...

Extra note!
Copying in workbench can be excruciatingly slow compared to todays standatds, so If you are going to copy over large amounts of data (several gigabytes), it can take several hours, and there's no progress bar or indication how much time is left *though you can kinda check it manually), so my advice is prepare and sort your files in windows, and do the copying overnight or while you're at work etc.

Note: This requires a windows computer and a functional WinUAE setup and assumes some basic knowledge of working with files and tranfers in workbench, and using winuae, and it requires atleast 10 gb of freely usable storage space on top of the size of amigavision
(This may also require you to modify amigavision workbench to accept drives over 4gb)

For other systems like Mac or linux, a capable amiga emulator should work as well, as long as you can mount hdf files and local folders, since most of the work is done in workbench.

### Aaaanyway, here's a 10 step basic guide to what i did:


1) Copy the "MegaAGS.hdf" and "MegaAGS-Saves.hdf" files to your computer or alternatively extract them from the amigavision archive. 
(If you are doing this while also upgrading to a new version of amigavision, replace the "MegaAGS.hdf" on your AP with the new one, and extract a new copy to work on, otherwise, make sure you only copy and don't move the original MegaAGS.hdf file as we need the one on the AP sdcard to be intact)

2) Start Winuae and mount "MegaAGS.hdf", "MegaAGS-Saves.hdf" and the local folder with all the files you want to tranfer.

Note! Depending on Workbench config etc, You may have to mount "MegaAGS-Saves.hdf" and your local folder together with "MegaAGS.hdf" in seperate batches, if this is the case, mount "MegaAGS-Saves.hdf" first and continue following the steps.
("MegaAGS-Saves.hdf" is typically mostly empty space, so it should only take a few seconds or minutes to transfer)

3) Launch into workbench (either with a separate hdf or the adf, don't boot amigavision) and open the "Amiga" and "Data" disk drawers.

4) Delete everything inside "Amiga" and "Data", but dont delete the "Trashcan" (if applicable, but do delete any files in the trashcan if there are any).
(Note! Don't format unless you know what you're doing, this method is the simplest way to ensure compatibility with filesystems etc)

5) Copy all the files from "MegaAGS-Saves.hdf" to the "Amiga" disk drawer, then copy over your main bulk of files to where you want them (but leave atleast 50-100mb free on the amiga drawer)

Note! If you are doing this in seperate batches, unmount "MegaAGS-Saves.hdf" now and return to step 2 and follow each step again, only with your local folder mounted instead.

6) Find something else to do while the files are being transferred, like taking up knitting, or learning the art of cat herding, or have a yelling contest with the neighbours dog, or sing Welsh folksongs at the bank, or shatpen your markers or become an international double agent, etc.

7) After everything has been tranferred over, make sure everything is where and how you want it to be.
If "MegaAGS-Saves.hdf" still mounted, unmount it now and move it to another folder or delete it.

8) Back in workbench, Rename the disk drawer "Data" to anything you want (except Data), and rename "Amiga" to "Saves" (this is mandatory)
(If you copied the contents of "MegaAGS-Saves.hdf" to another disk drawer, name that "Saves" instead)

9) Exit WinUAE and rename "MegaAGS.hdf" to "MegaAGS-Saves.hdf"

10) Copy "MegaAGS-Saves.hdf" to root:/Assets/amiga/common on your Analogue Pocket card.

You should now have a bigger "Save drive" as well as an additional drive cotaining all the content you copied over


Tl;dr: duplicate "MegaAGS.hdf", use winuae to empty drive and rename disk drawers, copy contents of "MegaAGS-Saves.hdf" to one of the disk drawers and remove, finally rename "MegaAGS.hdf" to "MegaAGS-Saves.hdf" and copy to sdcard

**Comment by limi** [comment]
> duplicate "MegaAGS.hdf", use winuae to empty drive and rename disk drawers, copy contents of "MegaAGS-Saves.hdf" to one of the disk drawers and remove, finally rename "MegaAGS.hdf" to "MegaAGS-Saves.hdf" and copy to sdcard

No need to make it so complicated if all you need is more space. As documented, there are multiple blank HDFs of various sizes included in `Extras.zip`, and you can simply replace `MegaAGS-Saves.hdf` with e.g. the 2GB blank HDF.
